modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -707,6 +709,58 @@ public void testInsertValueOverflow(String type, long max, long min) {
         }
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "'id1,id2', id1",
+            "'id1,id2', id2",
+            "'id1,id2', 'id1,id2'",
+            "'id1,id2', 'id2,id1'",
+            "'id2,id1', id1",
+            "'id2,id1', id2",
+            "'id2,id1', 'id1,id2'",
+            "'id2,id1', 'id2,id1'",
+    })
+    void test(String pkDefinition, String colocateByDefinition) {
```

Comment:
should we check the same for disabled rule for KV optimization?

Reply:
added 2 more test cases to verify multistep execution